### PR TITLE
[WIP] Added obstacle avoidance healthiness topic in vehicle status msg.

### PR DESCRIPTION
**Pull request by cmic0**
This commit adds to the pre-arm checks the avoidance healthiness check. Arming is temporary denied if obstacle avoidance is enabled but not yet running.
Added a print once flag to prevent excessive message spamming in QGC if avoidance system goes down.

Tested in SITL (Ubuntu 16.04).

As per discussion of future work:

- Maybe the defined flag could be used to refactor the failsafe triggering conditions in case the avoidance system stops mid-air?
(https://github.com/PX4/Firmware/blob/master/src/modules/mc_pos_control/mc_pos_control_main.cpp#L1205)
- Could even worth to add a parameter to specify the "timeout" time before trigger the "WAITING FOR AVOIDANCE SYSTEM" message (https://github.com/Auterion/Firmware/blob/7e6dce4bf1ee78d675594d7f28290637a3de4885/src/modules/commander/Commander.cpp#L3958 ) ? 
I think this parameter could be adjusted depending on the companion computer performances (i.e for slow booting platforms maybe 5s are not enough). This is meant to reduce warning messages in QGC.

### CODE REVIEW

File: src/modules/commander/Commander.cpp [hunk 1]
```diff
@@ -3950,19 +3952,26 @@ void Commander::data_link_check(bool &status_changed)
...
 			_avoidance_system_not_started = hrt_absolute_time();
-			mavlink_log_critical(&mavlink_log_pub, "AVOIDANCE SYSTEM DID NOT START");
+			mavlink_log_critical(&mavlink_log_pub, "WAITING FOR AVOIDANCE SYSTEM");
 		}
 
 		//if heartbeats stop
 		if ((_datalink_last_heartbeat_avoidance_system > 0)
 		    && (hrt_elapsed_time(&_datalink_last_heartbeat_avoidance_system) > 5_s) &&
 		    (hrt_elapsed_time(&_avoidance_system_lost) > 5_s)) {
 			_avoidance_system_lost = hrt_absolute_time();
```
Comment (mrivi) [inline src/modules/commander/Commander.cpp]: I know that it's not from this PR but a couple of lines above `_avoidance_system_lost`  is a boolean

File: src/modules/commander/Commander.cpp [hunk 2]
```diff
@@ -3950,19 +3952,26 @@ void Commander::data_link_check(bool &status_changed)
...
-			mavlink_log_critical(&mavlink_log_pub, "AVOIDANCE SYSTEM LOST");
+
+			if (!print_once) {
```
Comment (mrivi) [inline src/modules/commander/Commander.cpp]: I think there is no need of `print_once` if you set `_avoidance_system_lost=true`

**Comment by cmic0** [comment]
@mrivi to prevent merging conflicts I think we should define where to add the new topic and unify the naming convention. There is mainly one difference:
 - If the boolean flag for avoidance is added in vehicle_status, the message will be logged every 200ms.
-  If added in vehicle_status_flags will be logged at full rate (see [logger.cpp](https://github.com/PX4/Firmware/blob/master/src/modules/logger/logger.cpp#L657) )

I'd rather go with full rate logging.

File: src/modules/commander/Commander.cpp (hunk 1 above)
Comment (cmic0) [reply src/modules/commander/Commander.cpp]: I got tricked in the [header definition](https://github.com/PX4/Firmware/blob/99f8e38de02805723ce725c4299e35f5ee8cc4fa/src/modules/commander/Commander.hpp#L189) where is declared as uint64_t. I am gonna try tomorrow by cherry picking your commits.

**Comment by MaEtUgR** [comment]
Like just discussed with @dagar in a call the message `vehicle_status_flags` contains the status from various system components to be available to the state machine read-only and hence would be currently a good place to add an `external_avoidance_available` flag. On the long run we need to improve this architecture and the structure we work towards should be discussed in the dev call.

**Comment by cmic0** [comment]
I am going to resurrect this PR and rebase it on top of #11550.
The message flag will be refactored according to this comment https://github.com/PX4/Firmware/pull/11454#discussion_r258334874

**Comment by cmic0** [comment]
Closing this PR; I am going to create soon a new one based on #11550.

**Comment by cmic0** [comment]
Replaced by #11638

**Pull request closed without merging**